src/gluonts/itertools.py

```diff
@@ -289,6 +289,62 @@ def __iter__(self):
         return filter(self.fn, self.iterable)
 
 
+@dataclass
+class WeightedIterables:
+    """
+    Given a list of Iterables `iterables`, generate samples from them.
+
+    When `probabilities` is given, sample iteratbles according to it.
+    When `probabilities` is not given, sample iterables uniformly.
+
+    When one iterable exhausts, the sampling probabilities for it will be set to 0.
+
+        >>> from toolz import take
+        >>> from gluonts.itertools import Cyclic
```

Comment:
No need to import things from the same module.

Reply:
Also, don't think we need it for demonstrating what this does.

Reply:
I will remove the `Cyclic` import but leave the `toolz` import.